### PR TITLE
Converted LogEntry to a POJO.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: java/src/org/certificatetransparency/ctlog/LogEntry.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.certificatetransparency.ctlog;
+
+import org.certificatetransparency.ctlog.proto.Ct;
+
+
+public class LogEntry {
+  public Ct.X509ChainEntry x509Entry;
```
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: Do you plan next to POJOfy X509ChainEntry and PrecertChainEntry?

File: java/src/org/certificatetransparency/ctlog/LogEntry.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  public Ct.PrecertChainEntry preCertEntry;
```
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: case nit - it's 'precert' normally in our code, no?

File: java/src/org/certificatetransparency/ctlog/LogEntry.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  public LogEntry() {
```
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: perhaps this ctor can be omitted?

File: java/src/org/certificatetransparency/ctlog/LogEntry.java (hunk 1 above)
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: Yes, much POJOing. I'm momentarily stuck with `SignedCertificateTimestamp` and `DigitallySigned`, but I should be able to sort it out at some point...

Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: The end state is to not require `protobuf.jar` or `protoc` to build the Java side.

File: java/src/org/certificatetransparency/ctlog/LogEntry.java (hunk 2 above)
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: Except for `PreCert`?

The former protobuf setter was called `setPrecertEntry`, so you mean that this should be called `precertEntry`?

File: java/src/org/certificatetransparency/ctlog/LogEntry.java (hunk 3 above)
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: You can omit constructors? :smiley:

Comment (phad) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: It was a punt on my behalf, but this confirms it:
http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.8.9

File: java/src/org/certificatetransparency/ctlog/LogEntry.java (hunk 2 above)
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: yes, it should be 'precertEntry'

File: java/src/org/certificatetransparency/ctlog/LogEntry.java (hunk 1 above)
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: Cool, sgtm

File: java/src/org/certificatetransparency/ctlog/LogEntry.java (hunk 2 above)
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/LogEntry.java]: Oh, I remember, I was looking at the local variable that it was being set from. Changed it.

Of note, this field is write-only. :stuck_out_tongue:

**Pull request merged**